client/my-sites/domains/domain-management/list/bulk-all-domains.tsx

```diff
@@ -41,13 +42,24 @@ export default function BulkAllDomains( props: BulkAllDomainsProps ) {
 		<OptionsDomainButton key="breadcrumb_button_1" specificSiteActions allDomainsList />,
 	];
 
+	const domainsTableColumns: DomainsTableColumn[] = [
+		{
+			name: 'domain',
+			label: translate( 'Domain' ),
+			isSortable: true,
+			initialSortDirection: 'asc',
+			supportsOrderSwitching: true,
+			sortFunctions: [ getSimpleSortFunctionBy( 'domain' ) ],
+		},
+	];
+
 	return (
 		<>
 			<PageViewTracker path={ props.analyticsPath } title={ props.analyticsTitle } />
 			<Main wideLayout>
 				<BodySectionCssClass bodyClass={ [ 'edit__body-white' ] } />
 				<DomainHeader items={ [ item ] } buttons={ buttons } mobileButtons={ buttons } />
-				<DomainsTable domains={ domains } />
+				<DomainsTable domains={ domains } columns={ domainsTableColumns } />
```

Comment:
I'm wondering whether we really need to accept `columns` as a prop. We want to be able to share the `<DomainsTable>` component, but I can't imagine we want each place to provide their own columns?

The other thing to keep in mind is that not every column in the table is going to correspond with a field in the `DomainData` object. There's going to be columns with buttons for resolving errors. And there's the column with the ellipsis button too.

This approach does allow us to share the existing `getSimpleSortFunctionBy` function which is good. I think this functionality should be part of the `domains-table` package though. Someone else using the `<DomainsTable>` in another context is going to need to reimplement all the sorting themselves.

If we do want to keep the columns customisable, maybe the way to do that would be to have the `useDomainsTable` hook return the columns data. That's how I imagined allowing the user to customise things. The logic for fetching domains data (or in this case setting up columns) is still handled in the package. But if the user of the component wants to filter or modify that stuff before passing it to the component then they can do that.

Reply:
I see your point, It's probably best to encapsulate columns as part of the domains table and move the neccessary sorting utils to the package also. Maybe in the future there might be a need for customisable column but that seem like we should be changing the column in the domains-table package if that's the case.

Reply:
>I'm wondering whether we really need to accept columns as a prop. We want to be able to share the <DomainsTable> component, but I can't imagine we want each place to provide their own columns?

I resonate with that. I think that now that we have the `isAllSitesView` prop we can switch the columns according to that prop.

>Maybe in the future there might be a need for customisable column but that seem like we should be changing the column in the domains-table package if that's the case.

Yes 👍

Reply:
> The other thing to keep in mind is that not every column in the table is going to correspond with a field in the DomainData object. There's going to be columns with buttons for resolving errors. And there's the column with the ellipsis button too.

I was thinking about adding a `renderCell` function in `DomainsTableColumn` so a custom component can be rendered in that column cell if we like.